Expensify/App
# Incorporate Add Bank Account / KYC checks into Send Money + Pay request flow

cc @nickmurray47 

### Details

Tracking issue and additional context: https://github.com/Expensify/Expensify/issues/185320

This PR adds a way to effectively "block" a specific user action that requires a gold wallet and then to "continue" that action once they successfully make it through these flows. Since the flows all happen on different pages/screens it is tricky to enable the "continue" part of whatever user action they were taking. By using a "pay wall" style component we can save a reference to the component whenever it is mounted and call it's defined "continue" callback.

- [x] Holding on https://github.com/Expensify/App/pull/6992 as the `AdditionalDetailsStep` was recently broken and the test steps here can't be completed until this is fixed.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/187804
$ https://github.com/Expensify/Expensify/issues/187803

### Tests

1. Start with an account that has a SILVER wallet
2. Attempt to Send Money to someone else
3. Verify you are asked to add a payment method with a popover
4. Select bank account and verify you are brought to the Plaid flow and go through it
5. On success verify, you are next brought to the KYC flow and follow the second set of instructions here which explain how to test the ActivateWallet flow on dev -> https://stackoverflow.com/c/expensify/questions/10607
6. Once you successfully get through this flow the original action you wanted to take (send money) should succeed
7. Repeat with a new account but this time test paying a request
8. Logout after adding a bank account + passing KYC checks then log back in and verify that you do not hit the KYC wall anymore.

### QA Steps
Internal QA

### Tested On


### Screenshots


#### Web

![2022-01-07_11-36-01](https://user-images.githubusercontent.com/32969087/148610698-ef3abf6b-f90f-4a32-bd04-6caf05429403.png)

#### Mobile Web


#### Desktop


#### iOS

![2022-01-07_11-52-57](https://user-images.githubusercontent.com/32969087/148612452-3c1844e0-92ff-424d-a5f3-b67b27f74e75.png)
![2022-01-07_11-55-13](https://user-images.githubusercontent.com/32969087/148612471-aa9ce484-cb6a-4750-ad2a-d844acab92bf.png)

#### Android

![2022-01-07_12-07-05](https://user-images.githubusercontent.com/32969087/148614838-0adafba0-ffc0-4925-be82-cc0cef600798.png)
